### ISSUE TITLE
Design Intent - Accessibility Feedback - [govcio-vebt, Statement of Assurance of Compliance with 85% Enrollment Ratios (VA Form 22-10215), VA Form 22-10215

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#93568]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Thank you for walking through all of this, and being so open to questions and feedback! This is kind of an impossible task, and I appreciate the effort you've put into it.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [ ] **Consider:** On the [intro page](https://www.figma.com/design/f9TussyZYnC5UVSnqBPJqC/VEBT-Form-22-10215%3A-Statement-of-Assurance?node-id=249-27188&t=ifCZMMg7hrsrBabG-4), you start out with some very jargon-y blocks of text. For example: "Use this form (VA Form 22-10215) to provide 85/15 calculations as required by Title 38 United States Code (U.S.C.) 3680A(d) and 38 Code of Federal Regulations (CFR) 21.4201" etc. Since the audience for this page is university administrators and not Veterans this is a little less of a concern, but I would still encourage you to look for ways to follow [plain language guidelines](https://design.va.gov/content-style-guide/plain-language/) throughout the product wherever possible. University administrators are people too!
- [ ] **Must:** Whenever you have a table, the table needs to have a `caption` associated with it. Despite the name, the `caption` functions as a title of the table (and is styled that way in the VA design system).  The [`va-table` component](https://design.va.gov/storybook/?path=/docs/uswds-va-table--docs) does this out of the box with the `table-title` prop.
   
   In general, tables are more difficult for screen reader users to navigate and parse. Providing a caption/title gives screen reader users (and everyone, really) a summary of the table they've just encountered and the opportunity to decide whether or not to engage with that material or just skip over it. Otherwise, screen reader users have to listen cell-by-cell to enough cells to determine what the table is and whether it's something they want to explore further.
   
   So, for example, on the [intro page](https://www.figma.com/design/f9TussyZYnC5UVSnqBPJqC/VEBT-Form-22-10215%3A-Statement-of-Assurance?node-id=249-27188&t=ifCZMMg7hrsrBabG-4), a screen reader user may already know the due dates and may prefer to skip past that. Providing a table caption gives them the information they need to make that decision. Same goes for when users begin populating the data table they're submitting.
- [ ] **Should:** Ooof, those 85/15 calculation instructions! There's a lot there! But since the audience for this form is generally going to be familiar with these calculations, I don't think the instructions need to be presented on this page --- and presenting them inside an accordion that you have to scroll back up to, open, navigate to the section you want, then navigate back to the input you started at is a heavy cognitive load and a tricky task for screen reader users.
   
   What I would recommend instead: The calculation instructions should live on their own separate page, outside of the form flow. Helper text in close proximity to each input could then link (opening in a new tab) directly to the section of the instructions that's relevant to that input --- something like:
   ```
   Number of non-supported students FTE (*Required)
   <a href="https://va.gov/some-url/some-slug#number-non-supported" target="_blank">How to calculate the number of non-supported students FTE (opens in a new tab)</a>
   [input]
   ```
   Note that I wouldn't recommend linking to the instructions in the PDF version of the form. Better to create a proper HTML page, which is easier to link to individual sections and generally a more accessible experience.
   
   Where appropriate, you may also be able to put shorter instructions in the `hint` text of an input without having to link to the instructions page.
- [ ] **Should:** I'd encourage you to follow the [multi-page list and loop pattern](https://design.va.gov/patterns/ask-users-for/multiple-responses#how-to-design-and-build---multi-page) as much as is possible:
   - User starts with a page with just the program information inputs.
   - After the first entry is submitted, take the user to a page with a success message, the table is displayed with one entry, and there's a "Do you have another...?" question with yes/no options.
   - Answering yes takes you to back to the page with just the program information inputs.
   - After that entry is submitted, take the user back to the success message, the table (now two entries), and the "Do you have another...?" question.
   
   Here's what this gets you:
   - Focused, task-specific, lighter-weight pages. When you're adding an entry, you're only adding an entry. When you're reviewing what you've entered so far, you're only doing that and deciding if you have more. That's hopefully better for cognitive load and should be significantly easier for screen reader users to navigate.
   - Success messaging. The success alert tells users that their action completed successfully, without relying on visual cues of something changing in the table --- or even not having any visual cues, once the table starts paginating.
   - A yes/no decision to keep adding at every step, as opposed to the [modal interruption](https://www.figma.com/design/f9TussyZYnC5UVSnqBPJqC/VEBT-Form-22-10215%3A-Statement-of-Assurance?node-id=358-35119&t=ifCZMMg7hrsrBabG-4) when you attempt to proceed. We use that kind of modal interruption for destructive actions (are you sure you want to remove that?) and I suspect that interrupting users when they've successfully completed a task is going to go against that mental model.
   
   This multi-page list-and-loop pattern tests well in user research. But none of that research involved the kind of bulk entry of data that's happening here. It may still work well, but as always listen to real live users and adjust based on that feedback!
- [ ] **Should:** On [Page 17](https://www.figma.com/design/f9TussyZYnC5UVSnqBPJqC/VEBT-Form-22-10215%3A-Statement-of-Assurance?node-id=274-30808&t=ifCZMMg7hrsrBabG-4) of the Figma file, for the "Total number of supported students enrolled" input, I would recommend including the info alert for when the value is less than 10 as hint text that's always visible, rather than conditionally revealing it. I don't think hiding that information really helps anyone, and you often need to do extra work to ensure screen reader users are aware of conditionally-revealed text like that. I don't think the work is really worth the effort if it can just be visible hint text.
- [ ] **Must:** In the data table, deleting an entry is an action and not a destination, so it should be a button rather than a link. The [multiple page list-and-loop](https://design.va.gov/patterns/ask-users-for/multiple-responses#how-to-design-and-build---multi-page) uses the [Button - Icon - Delete variation](https://design.va.gov/storybook/?path=/docs/components-va-button-icon--docs#delete), which I think would be appropriate here as well.
- [ ] **Consider:** With list-and-loops like this, we typically provide users with a way to edit entries as well as delete them. At the scale of what a user is going to be entering here... I'm not sure how important that is. But I think it's worth spending some time exploring what that might look like (eg. edit and delete in a single "Actions" column rather than a "Delete" column).
- [ ] **Must:** Whenever you have repeated button or link text, those elements need to have unique accessible names so that screen reader users can successfully identify what the button is acting on/where the link goes. This is typically accomplished using an `aria-label`, which doesn't change the visible text but provides text to be announced just to screen reader users.
   
   For the delete actions, an `aria-label` of "Delete [program name]" seems like it would work here. You know the data you're working with here better than me, but you're looking for something concise and relatively unique from one or more of the fields that the user is providing for each entry. You don't need to be too verbose ("delete astronomy program with 1000 total students...") but you do need to reduce the likelihood of collisions with other entries.
   
   This is also something to watch for with the "How is this calculated?" additional info components. Something like "How is the supported student percentage calculated" or similar would be fine.
- [ ] **Must:** This is something to explore with your engineering team --- make sure that the PDF that's generated at the end of the form flow is itself an accessible PDF.
- [ ] **General concern:** As discussed in the meeting this is outside of the scope of your team's work, but I'm just documenting here that some accessibility assessment should be done for the upload portal. If a school official with a disability gets through this form flow but then can't successfully upload their submission, that's still a breakdown in the overall process.

## Governance team actions
- [ ] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [ ] Add the VFS team product label
- [ ] Add the VFS team the feature label (if applicable)
- [ ] Add the touchpoint labels
- [ ] Add the practice area labels
- [ ] Add the Collaboration Cycle initiative milestone

**Comment by ameliahackingVA** [comment]
All feedback has been incorporated into the form.
